api/src/services/listing.service.ts

```diff
@@ -1753,4 +1758,47 @@ export class ListingService implements OnModuleInit {
       success: true,
     };
   }
+
+  /**
+    runs the job to auto expire lotteries that are passed their due date
+    will call the the cache purge to purge all listings as long as updates had to be made
+  */
+  async expireLotteries(): Promise<SuccessDTO> {
+    if (process.env.LOTTERY_DAYS_TILL_EXPIRY) {
+      this.logger.warn('changeExpiredLotteryStatusCron job running');
+      await this.markCronJobAsStarted(LOTTERY_CRON_JOB_NAME);
+      const expiration_date = new Date();
+      expiration_date.setDate(
+        expiration_date.getDate() -
+          Number(process.env.LOTTERY_DAYS_TILL_EXPIRY),
+      );
```

Comment:
We use dayjs in the system to do date manipulation. ` dayjs(new Date()).subtract(Number(process.env.LOTTERY_DAYS_TILL_EXPIRY), 'days').toDate()`